### ISSUE TITLE
The output can not work.

### ISSUE THREAD

**Issue by CreateAndInject** [HoLLy-HaCKeR/EazFixer#10]
[1.zip](https://github.com/HoLLy-HaCKeR/EazFixer/files/2269411/1.zip)

if (type.IsGenericType)  //type is null
{
	flag2 = false;
}

It seems this method is generated by eaz, your tool doesn't clean completely

whole code
```
// #=qhV51mqsmZyKWrUetPxG5f8m$h3XhMl0AIaoJwVc$D1E=
[DebuggerNonUserCode]
private MethodBase #=z8v9YWyfC6tuClV80LCgyA_1FwG3San9xrA==(int #=zrn90pR8=, #=qKwaTyl6Rv9mcRmvdxY4NXmcIEkJcl0OEF11sG53sWvk= #=zCl9$JzU=)
{
	Dictionary<int, object> dictionary = #=qhV51mqsmZyKWrUetPxG5f8m$h3XhMl0AIaoJwVc$D1E=.#=z8vSQRPg=;
	Dictionary<int, object> dictionary2;
	if (3 != 0)
	{
		dictionary2 = dictionary;
	}
	object obj = dictionary2;
	if (true)
	{
		Monitor.Enter(obj);
	}
	MethodBase result;
	try
	{
		bool flag = true;
		bool flag2;
		if (6 != 0)
		{
			flag2 = flag;
		}
		object obj2;
		if (flag2 && #=qhV51mqsmZyKWrUetPxG5f8m$h3XhMl0AIaoJwVc$D1E=.#=z8vSQRPg=.TryGetValue(#=zrn90pR8=, out obj2))
		{
			MethodBase methodBase = (MethodBase)obj2;
			if (2 != 0)
			{
				result = methodBase;
			}
		}
		else if (#=zCl9$JzU=.#=zX1wW8IsArNpN_oBDfg==() == 1)
		{
			MethodBase methodBase2 = this.#=zf2bab1g=.ResolveMethod(#=zCl9$JzU=.#=zrIrFLDNPcENt9pkCVvza8mL0m1p0());
			MethodBase methodBase3;
			if (true)
			{
				methodBase3 = methodBase2;
			}
			if (flag2)
			{
				#=qhV51mqsmZyKWrUetPxG5f8m$h3XhMl0AIaoJwVc$D1E=.#=z8vSQRPg=.Add(#=zrn90pR8=, methodBase3);
			}
			MethodBase methodBase4 = methodBase3;
			if (5 != 0)
			{
				result = methodBase4;
			}
		}
		else
		{
			#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo= _#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo= = (#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=)#=zCl9$JzU=.#=z2m1nqKAQIf_jctR6PdUesuijy4gxvv9CZIRuQFqHSCty();
			#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo= _#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=2;
			if (2 != 0)
			{
				_#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=2 = _#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=;
			}
			if (_#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=2.#=zuc1ElAJDK6tNfflOZNdk01A=())
			{
				result = this.#=z3ArTbSmHWfuj5Q64DCej4Sshg_3p(_#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=2);
			}
			else
			{
				Type type = this.#=ziG5bIhoUvy8pBi1kvJtL67DfE0AX573nXuXvc35gLmcw(_#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=2.#=zDV1gSv8XgtF_MjvpkH0e08Q=().#=zrIrFLDNPcENt9pkCVvza8mL0m1p0(), false);
				Type type2 = this.#=ziG5bIhoUvy8pBi1kvJtL67DfE0AX573nXuXvc35gLmcw(_#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=2.#=zoVLjew9ZtCpdRsundXmEQWKvSguy().#=zrIrFLDNPcENt9pkCVvza8mL0m1p0(), true);
				Type[] array = new Type[_#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=2.#=zRZYXzc5RlHjvU$D1BQ==().Length];
				for (int i = 0; i < array.Length; i++)
				{
					array[i] = this.#=ziG5bIhoUvy8pBi1kvJtL67DfE0AX573nXuXvc35gLmcw(_#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=2.#=zRZYXzc5RlHjvU$D1BQ==()[i].#=zrIrFLDNPcENt9pkCVvza8mL0m1p0(), true);
				}
				if (type.IsGenericType)
				{
					flag2 = false;
				}
				if (_#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=2.#=zV$HlPelviP3L6394IwTa60IKPn0D() == ".ctor")
				{
					ConstructorInfo constructor = type.GetConstructor(BindingFlags.Instance | BindingFlags.Public | BindingFlags.NonPublic, null, CallingConventions.Any, array, null);
					if (constructor == null)
					{
						throw new Exception();
					}
					if (flag2)
					{
						#=qhV51mqsmZyKWrUetPxG5f8m$h3XhMl0AIaoJwVc$D1E=.#=z8vSQRPg=.Add(#=zrn90pR8=, constructor);
					}
					result = constructor;
				}
				else
				{
					BindingFlags bindingAttr = #=qhV51mqsmZyKWrUetPxG5f8m$h3XhMl0AIaoJwVc$D1E=.#=za_3ai5XgP4YmJt4xaj9EeXILLULi(_#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=2.#=z_SqDH4WDUufAqV3c5AegaJxjWWggFlUHFvQ8RDw=());
					MethodBase methodBase5 = null;
					try
					{
						methodBase5 = type.GetMethod(_#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=2.#=zV$HlPelviP3L6394IwTa60IKPn0D(), bindingAttr, null, CallingConventions.Any, array, null);
					}
					catch (AmbiguousMatchException)
					{
						foreach (MethodInfo methodInfo in type.GetMethods(bindingAttr))
						{
							if (!(methodInfo.Name != _#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=2.#=zV$HlPelviP3L6394IwTa60IKPn0D()) && methodInfo.ReturnType == type2)
							{
								ParameterInfo[] parameters = methodInfo.GetParameters();
								if (parameters.Length == array.Length)
								{
									bool flag3 = false;
									for (int k = 0; k < array.Length; k++)
									{
										if (parameters[k].ParameterType != array[k])
										{
											flag3 = true;
											break;
										}
									}
									if (!flag3)
									{
										methodBase5 = methodInfo;
										break;
									}
								}
							}
						}
					}
					if (methodBase5 == null)
					{
						throw new Exception(string.Format("Cannot bind method: {0}.{1}", type.Name, _#=q0p7SnAY3GqJTIa5Cdjeux8Ehu4Rhzu_baBrMQqvVcNo=2.#=zV$HlPelviP3L6394IwTa60IKPn0D()));
					}
					if (flag2)
					{
						#=qhV51mqsmZyKWrUetPxG5f8m$h3XhMl0AIaoJwVc$D1E=.#=z8vSQRPg=.Add(#=zrn90pR8=, methodBase5);
					}
					result = methodBase5;
				}
			}
		}
	}
	finally
	{
		Monitor.Exit(dictionary2);
	}
	return result;
}
```

**Comment by PuffingIn2D** [comment]
Your program appears to be packed with virtualization, Please check my fork or check my hollypatch pull request or branch, The program should than run but will not be devirtualized.

**Comment by CreateAndInject** [comment]
1 What is devirtualized?
  It means it isn't unpacked? 

2 The output can not run also with different exception.

**Comment by HoLLy-HaCKeR** [comment]
EazFixer cannot devirtualize methods, and I'm not planning to implement it any time soon. I've created an issue #13 that is most likely what's causing your problem.

**Comment by HoLLy-HaCKeR** [comment]
Fixed since #15, you can now use `--fix-virt` or `--keep-types`. Please reopen this issue if the problem persists (make sure you deobfuscate the control flow first.)
